src/qutip_qip/qiskit/backend.py

```diff
@@ -0,0 +1,199 @@
+from email.header import Header
+from qutip import tensor, basis
+import numpy as np
+import uuid
+import random
+from qutip_qip import circuit
+
+from qutip_qip.operations.gateclass import Z
+from qutip_qip.circuit import QubitCircuit
+from qutip_qip.circuit.circuitsimulator import CircuitResult
+
+from .job import Job
+from .converter import convert_qiskit_circuit
+from qiskit.providers import BackendV1, Options
+from qiskit.providers.models import (
+    BackendConfiguration,
+    QasmBackendConfiguration,
+)
+from qiskit.result import Result, Counts
+from qiskit.result.models import ExperimentResult, ExperimentResultData
+from qiskit.quantum_info import Statevector
+from qiskit.circuit import QuantumCircuit
+from qiskit.qobj import QobjExperimentHeader
+
+
+class QiskitSimulatorBase(BackendV1):
+    """
+    The base class for qutip_qip based qiskit backends.
+    """
+
+    def __init__(self, configuration=None, provider=None, **fields):
+        pass
+
+    def run(self, qiskit_circuit: QuantumCircuit, **backend_options) -> Job:
+        """
+        Simulates a circuit on the required backend.
+
+        Parameters
+        ----------
+        qiskit_circuit : QuantumCircuit
+            The qiskit circuit to be simulated.
+
+        Result
+        ------
+        qutip_qip.qiskit.job.Job
+            Job that stores results and execution data
+        """
+        qutip_circ = convert_qiskit_circuit(qiskit_circuit)
+        job_id = str(uuid.uuid4())
+        job = Job(
+            backend=self,
+            job_id=job_id,
+            result=self._run_job(job_id, qutip_circ),
+            warnings=qutip_circ._warnings
+            if hasattr(qutip_circ, "_warnings")
+            else [],
+        )
```

Comment:
This part is a bit strange, the warning message should not be part of the circuit. I understand that you want to pass the message from the converter up to this level. But hiding it as an attribute looks very confusing.

There are standard ways to catch a warning message. Just like `try` `except` for the error message, you can also get write a `with` condition that catches warnings:
```python
with warnings.catch_warnings(record=True) as w:
    pass
```
But I think this is also not the best way to handle this here.

Maybe a better way is to the following:
- define a qutip-backend-unique configuration key e.g. `allow_custom_gate` in the configuration with the default `True`.
- Add a key-word parameter `allow_custom_gate=True` to `convert_qiskit_circuit`.
- When converting the circuit here, we read from the `BackendConfiguration` (There should be a way to access it here.) and pass it to `convert_qiskit_circuit`.
- If the custom gate is allowed. Use it without warning. Otherwise, we raise an *error*.

I think this makes sense especially because, for the pulse-level processor, we cannot use any custom gates.